File: shop-lite.Server/Models/Product.cs
public class Product
{
    public Guid Sku { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public int Stock { get; set; }
    public decimal Price { get; set; }
    public decimal? SalePrice { get; set; }
}

File: frontend/lib/schemas/product.ts
import { z } from 'zod'

export const ProductSchema = z.object({
  sku: z.string().uuid(),
  name: z.string(),
  description: z.string(),
  imageUrl: z.string().nullable(),
  stock: z.number().int(),
  price: z.number(),
  salePrice: z.number().nullable(),
})

export type Product = z.infer<typeof ProductSchema>
